docs: discourage the use of SQLite (#23794)
apache/superset: docs/docs/installation/configuring-superset.mdx

### Using a production metastore

By default, Superset is configured to use SQLite, which is a simple and fast way to get started
(without requiring any installation). However, for production environments,
using SQLite is highly discouraged due to security, scalability, and data integrity reasons.

> It's important to use only the supported database engines and consider using a different

Reviewer: ```suggestion
It's important to use only supported database engines and consider using a different
```

> It's important to use only the supported database engines and consider using a different
> database engine on a separate host or container.

Reviewer: ```suggestion
We recommend using another supported database engine on a separate host or container.
```

Author: shorter but seems a bit confusing to say "another supported database engine on a separate host"
